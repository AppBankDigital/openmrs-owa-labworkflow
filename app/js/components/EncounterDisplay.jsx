import React, { PureComponent } from 'react';
import R from 'ramda';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import moment from 'moment';
import { injectIntl } from 'react-intl';


class EncounterDisplay extends PureComponent {
  render() {
    const {
      order: {
        labResult = null,
      },
      type,
      intl,
    } = this.props;

    if (!R.isNil(labResult)) {
      if (type === "status") {
        return (
          <span>
            { labResult.resultStatus }
          </span>
        );
      }

      if (type === "collectionDate") {
        const statusesWithoutEncounter = [
          intl.formatMessage({ id: "app.labResult.status.ordered", defaultMessage: "Ordered" }),
          intl.formatMessage({ id: "app.labResult.status.canceled", defaultMessage: "Canceled" }),
          intl.formatMessage({ id: "app.labResult.status.expired", defaultMessage: "Expired" })
        ];
        if (!statusesWithoutEncounter.includes(labResult.resultStatus)) {
          return (
            <div className="table_cell test-type">
              { R.path(['encounter', 'encounterDatetime'], labResult) ? moment(labResult.encounter.encounterDatetime).format("DD-MMM-YYYY") : ""}
            </div>
          );
        }
        return null;
      }
    }

    return (
      <div className="spiner" />
    );
  }
}

EncounterDisplay.propTypes = {
  order: PropTypes.shape({}).isRequired,
  type: PropTypes.string.isRequired,
};

const mapStateToProps = ({
  labOrders: { orders },
}, {
  orderUUID,
}) => ({
  order: orders.find(item => item.uuid === orderUUID),
});

export default connect(mapStateToProps)(injectIntl(EncounterDisplay));
