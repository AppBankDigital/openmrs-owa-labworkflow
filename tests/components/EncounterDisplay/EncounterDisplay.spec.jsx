import React from 'react';
import { Provider } from 'react-redux';
import { mountWithIntl } from '@openmrs/react-components';
import EncounterDisplay from '../../../app/js/components/EncounterDisplay';


let mountedComponent;


const getComponent = (state, props) => {
  const store = mockStore(state);

  if (!mountedComponent) {
    mountedComponent = mountWithIntl(
      <Provider store={store}>
        <EncounterDisplay {...props} />
      </Provider>
      ,
    );
  }

  return mountedComponent;
};

describe('EncounterDisplay Component', () => {

  it('should render for order value of type collectionDate for status of Reported results', (done) => {
    const props = {
      orderUUID: 'some-order-uuid',
      type: 'collectionDate',
    };

    const state = {
      labOrders: {
        orders: [
          {
            uuid: 'some-order-uuid',
            labResult: {
              resultStatus: "Reported",
              encounter: {
                uuid: 'some-encounter-uuid',
                encounterDatetime: '12/2/2018',
              },
            },
          },
        ],
      },
    };

    const component = getComponent(state, props);
    expect(component).toMatchSnapshot();
    done();
  });

  it('should render for order value of type status', (done) => {
    const props = {
      orderUUID: 'some-order-uuid',
      type: 'status',
    };

    const state = {
      labOrders: {
        orders: [
          {
            uuid: 'some-order-uuid',
            labResult: {
              encounter: {
                uuid: 'some-encounter-uuid',
                encounterDatetime: '12/2/2018',
              },
            },
          },
        ],
      },
    };
    const component = getComponent(state, props);
    expect(component).toMatchSnapshot();
    done();
  });
});