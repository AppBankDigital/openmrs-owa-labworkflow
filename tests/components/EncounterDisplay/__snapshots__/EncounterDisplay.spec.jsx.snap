// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EncounterDisplay Component should render for order value of type collectionDate for status of Reported results 1`] = `
<Provider
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(EncounterDisplay)
    orderUUID="some-order-uuid"
    type="collectionDate"
  >
    <EncounterDisplay
      dispatch={[Function]}
      order={
        Object {
          "labResult": Object {
            "encounter": Object {
              "encounterDatetime": "12/2/2018",
              "uuid": "some-encounter-uuid",
            },
            "resultStatus": "Reported",
          },
          "uuid": "some-order-uuid",
        }
      }
      orderUUID="some-order-uuid"
      type="collectionDate"
    >
      <div
        className="table_cell test-type"
      >
        02-Dec-2018
      </div>
    </EncounterDisplay>
  </Connect(EncounterDisplay)>
</Provider>
`;

exports[`EncounterDisplay Component should render for order value of type status 1`] = `
<Provider
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(EncounterDisplay)
    orderUUID="some-order-uuid"
    type="collectionDate"
  >
    <EncounterDisplay
      dispatch={[Function]}
      order={
        Object {
          "labResult": Object {
            "encounter": Object {
              "encounterDatetime": "12/2/2018",
              "uuid": "some-encounter-uuid",
            },
            "resultStatus": "Reported",
          },
          "uuid": "some-order-uuid",
        }
      }
      orderUUID="some-order-uuid"
      type="collectionDate"
    >
      <div
        className="table_cell test-type"
      >
        02-Dec-2018
      </div>
    </EncounterDisplay>
  </Connect(EncounterDisplay)>
</Provider>
`;
